### ISSUE TITLE
Feature: mark played songs in the palette playlist

### ISSUE THREAD

**Issue by Gero5** [mpogue2/SquareDesk#958]
Is it possible to mark the songs played in a session in the palette's playlist.
greyin / highlighting / put an '*' before it or something like that. 

I do not play the songs from # 1 to # 8 instead I pick one from the list. I have two playlist one singer and one patter calls. if played titles are marked that would be helpful .

**Comment by mpogue2** [comment]
Ah, if I understand you, I think you're using playlists in this use case like what some other apps call "crates".  Remember those wooden slat-boxes that we used to keep vinyl records in?  It's a set of records, or in this case a set of related tracks/songs (could be for example, "Christmas Singers").  

That's a somewhat different use case than the original playlist idea, but I think it's a valid one.  It adds a bunch of complexity though (need to be able to clear these too, for example...).  I like the idea, but I think there's more definition work to be done here.

In this context, we should also probably talk about the definition of "played titles".  That has turned out to be somewhat tricky. Is it a) "was loaded" or b) "was played (for any amount of time)" or c) "was played (for more than N seconds in a row)" or d) "was played (for more than N seconds total, not counting pauses, before some other track was loaded".  

This comes up in the context of "songs played" at a dance, which I think in Germany you are tracking for copyright purposes, right?  I track this as well, but the definition of "played" I don't think we have right yet (it's currently "played for any amount of time").

**Comment by danlyke** [comment]
Would it be helpful to expand the notion of sessions somehow? Like make it
easier to create them and combine them?

I'm pondering this, and trying to figure out the use case, but I'm thinking
it's kind of a "these are the songs I want to use tonight, and have I
already played this one tonight?" sort of thing?

**Comment by mpogue2** [comment]
Yeah, Sessions don't work well for me right now, because I have Jokers from 6-9PM on Wednesdays, but it's sometimes 6:30-9PM, and we sometimes have Saturday sessions from 2:30-5 (but not every Saturday).  So an algorithmic Session definition is a great idea in theory, but in practice it doesn't work well for me.

So much so that tracking which songs have been played I do manually, using Export Song History, edit the CSV file, and then run an external program to tell me which songs I haven't played for a while.

So, the "track has been played" definition is an open question, I think, but also "track has been played for XYZ group" is an open Sessions question as well!

I'd love to have an automated way to do this (along the lines of SquareDesk philosophy = Automate All The Things), but is an automated mechanism even possible?  How does SquareDesk know (automatically, magically) who I'm playing a track for?

**Comment by Gero5** [comment]
proposal for auto detect sessions see #961

**Comment by Gero5** [comment]
> So, the "track has been played" definition is an open question

for me mark the song as soon as I double klick on it on playlist would do.

**Comment by mpogue2** [comment]
@Gero5 That doesn't work very well for me, because I often load a piece of music for just a few seconds (e.g. to check the singing call figure), and I don't want that load to count as "I played the song".  

Would these loads (perhaps even without any playback) count as "performance" for the German copyright reporting purposes?

**Comment by Gero5** [comment]
@danlyke 
> I'm thinking
> it's kind of a "these are the songs I want to use tonight, and have I
> already played this one tonight?" sort of thing?

exactly. my playlist are an unsorted collection of the songs I want to play. like records in a tool box in the old days.
with the new frames I have one list for patter music (palette # 1) and one for singers (palette # 2) and I pick what I think fits best to the mood on the dance floor. I never play the songs in the order on the playlist.

**Comment by Gero5** [comment]
> Would these loads (perhaps even without any playback) count as "performance" 
> for the German copyright reporting purposes?

due to the German copyright rules everything that is played **longer that 15 seconds** counts. 
Mark as played after 15 seconds playtime. That would be a good choice for me as well.

BTW: how is that with the song play history? If you load a song and never click on play is it on the list?

**Comment by mpogue2** [comment]
If it is loaded but not played, it currently doesn't go into the Song Play History.
If it is played at all, even for <1 second, it goes into the Song Play History (_song_plays_ table in the DB).

At some point, I would like to keep track of how long songs are played, so I can filter out the 5 second playback segments.  That was the point of #918 (oh, I forgot that I asked you the same question in #918 -- thanks for answering twice!)...

**Comment by Gero5** [comment]
in #918 you said
> "used" has a specific meaning, not just "played sometime on that date for a few seconds".)

is a 15 seconds hurdle helpful for that? then go for it. That would be the first time the German GEMA is helpful and not causing problems. :-)

**Comment by danlyke** [comment]
So just brainstorming: What it a playlist could be associated with a
session? Load a playlist, select that session?

(I currently end up manually selecting my sessions because I'm calling
fill-ins and such, no regular gigs.)

(I Message ID: ***@***.***>

>

**Comment by mpogue2** [comment]
@danlyke interesting idea...I do name all my playlists like this: groupname_date.csv, e.g. "jokers_2023.11.15.csv".  That allows us to aggregate by groupname ("jokers" in this case).

I'm not sure how this will change for people who display 3 playlists at a time in the new UX... That might be a different use case...

**Comment by mpogue2** [comment]
Epic is now #1721.
